Libraries/NativeAnimation/Drivers/RCTFrameAnimation.m

```diff
@@ -95,11 +101,19 @@ - (void)stepAnimationWithTime:(NSTimeInterval)currentTime
   NSUInteger nextIndex = startIndex + 1;
 
   if (nextIndex >= _frames.count) {
-    // We are at the end of the animation
-    // Update value and flag animation has ended.
-    NSNumber *finalValue = _frames.lastObject;
-    [self updateOutputWithFrameOutput:finalValue.doubleValue];
-    _animationHasFinished = YES;
+    if (_iterations == -1 || _currentLoop < _iterations) {
+      // Looping, reset to the first frame value.
+      _animationStartTime = currentTime;
+      _currentLoop++;
+      NSNumber *firstValue = _frames.firstObject;
+      [self updateOutputWithFrameOutput:firstValue.doubleValue];
+    } else {
+      _animationHasFinished = YES;
```

Comment:
Does it matter that you are re-ordering this before `[self updateOutputWithFrameOutput:finalValue.doubleValue]`?